root/etc/e-smith/templates/etc/ssh/sshd_config/45AllowGroups2Sshd

```diff
@@ -0,0 +1,55 @@
+{
+    # Allow groups to sftp/ssh with different policies
+    #
+    # if $sssd{ShellOverrideStatus} is disabled we are in legacy (everybody is allowed with a shell access)
+    # if $sssd{ShellOverrideStatus} is enabled we allow to root and members of domain admins group
+    # if $sssd{ShellOverrideStatus} is enabled  with non empty $sshd{AllowGroups} 
+    #   we allow root, members of domain admins group and members of groups inside the AllowGroups prop
+    #
+    # members restricted to sftp (SftpRestrictedGroups) must be allowed to sshd also
+
+    use strict;
+    use esmith::ConfigDB;
+    use esmith::AccountsDB;
+    my $db = esmith::ConfigDB->open_ro();
+
+
+    sub uniq {
+        my %seen;
+        return grep { !$seen{$_}++ } @_;
+    }
+
+    my $domain = $db->get_value('DomainName')|| die ('Cannot retrieve DomainName');
+    my $PermitRootLogin = $db->get_prop('sshd','PermitRootLogin') || "no";
+    my $policy = $db->get_prop('sssd','ShellOverrideStatus') || "disabled";
+
+    my @AllowGroups = ();
+    foreach ( split(',',$db->get_prop('sshd','AllowGroups') || '')) {
+        my ($group, $sftp) = split(':', $_);
+
+        if(!$group) {
+            next;
+        }
+        push @AllowGroups, $group;
```

Comment:
Do we need the `@domain` suffix here too?
```suggestion
        $group .= "\@$domain" if ($group !~ '@');
        push @AllowGroups, $group;
```

Reply:
yes we need but it is a matter how the UI will write to the DB, to allow the group famille

`AllowGroups root "domain admins@nethservertest.org" famille@nethservertest.org`

Reply:
however, I thing I did a mistake, we need to retrieve the group name from /usr/libexec/nethserver/list-groups if we are bound to a remote ldap or running a local LDAP. 

The esmith $DomainName could be different of the group@domainname of the remote account provider

Reply:
> yes we need but it is a matter how the UI will write to the DB, to allow the group famille

Yes, maybe redundant, but doesn't harm. And the `cockpit.socket/delegation` prop uses the short form. I prefer it.

> The esmith $DomainName could be different of the group@domainname of the remote account provider

Look at `list-groups` code. The domain suffix is the output of `hostname -d`, no matter the AD  domain is.

I prefer the short form in the DB. Even if we implement a multi-tenant multi-domain solution the short form can be easily mapped to a "default/primary" domain. And if we implement a change  of domain name, there's no need to fix the DB contents.

Reply:
yes the short group name is fine for me, I missed this information
> Look at list-groups code. The domain suffix is the output of hostname -d, no matter the AD domain is.

It will make my life easier

Reply:
done, we keep only short group name and we add the domainName